### PR TITLE
Fix `Grid` rendering glitches when used inside `Panel` with `ModalSup…

### PR DESCRIPTION
**Pull request by ghsolomon**
…port`

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/grid/Grid.js [hunk 1]
```diff
@@ -552,6 +557,20 @@ class GridLocalModel extends HoistModel {
...
+     * Force Grid to redraw rows when switching between inline and modal views
+     * {@see modalSupport}
+     */
+    modalReaction() {
+        const modalSupportModel = this.lookupModel(ModalSupportModel);
+        if (!modalSupportModel) return null;
+
+        return {
+            track: () => modalSupportModel.isModal,
+            run: () => wait().then(() => this.model.agApi.redrawRows())
```
Comment (lbwexler) [inline cmp/grid/Grid.js:570]: can also just use `delay: 0` on reaction for slightly flatter code

Comment (ghsolomon) [reply cmp/grid/Grid.js]: Needed to do `delay: 1` (maybe MobX makes `delay: 0` a no-op?), but that worked and is cleaner

**Comment by lbwexler** [comment]
got it!  could do debounce: 0, also, that works and perhaps more consistent.

I think the comment is not needed here as well

**Comment by ghsolomon** [comment]
`debounce: 0` works.  I wonder if `delay` is clearer since we really are intending to delay the reaction in this case?  Not really worrying about it firing multiple times in succession.  But happy to update (and remove the comment either way)!

File: cmp/grid/Grid.js [hunk 2]
```diff
@@ -177,6 +178,10 @@ class GridLocalModel extends HoistModel {
         this.addReaction(this.sizingModeReaction());
         this.addReaction(this.validationDisplayReaction());
 
+        // If ancestor of a modalSupport cmp:
```
Comment (lbwexler) [inline cmp/grid/Grid.js:181]: no need for comment here,  the method itself is perfectly clear, and the benefits of having the method mean we don't need to distract in the constructor.

**Pull request merged**